### PR TITLE
Fix dom select[All]

### LINKED ISSUE #368: Can't listen/trigger events on the `window` object
`dom.select` and `dom.selectAll` check for `selector.nodeType` which is `undefined` for the `window` object. This makes listening/triggering events on the `window` impossible.

In addition to that there is a bug in [`trigger`](https://github.com/flightjs/flight/blob/master/lib/base.js#L104).
When triggering with an event object `element` ends up being `this.node` and `event` = `selector/node`
Tests aren't failing because we assume that the first parameter is a string e.g. `document`, however if you try to trigger on an actual dom element it fails.

### PR DESCRIPTION
**Pull request by giuseppeg**
Fixes #368

### CODE REVIEW

File: test/spec/events_spec.js [hunk 1]
```diff
@@ -221,6 +221,29 @@ define(['lib/component', 'lib/registry', 'lib/dom'], function (defineComponent,
...
+        'body'
+      ];
+
+      specialElements.forEach(function (element) {
+        var spy = jasmine.createSpy(element.toString());
+        instance2.on(element, 'click', spy);
+        instance1.trigger('click');
+        instance2.off(element, 'click');
+        instance1.trigger('click');
+        expect(spy.callCount).toBe(1);
```
Comment (giuseppeg) [inline test/spec/events_spec.js]: we are using an old version of jasmine :(

File: test/spec/events_spec.js [hunk 2]
```diff
@@ -221,6 +221,29 @@ define(['lib/component', 'lib/registry', 'lib/dom'], function (defineComponent,
...
+        document.documentElement,
+        'html',
+        'body'
+      ];
+
+      specialElements.forEach(function (element) {
+        var spy = jasmine.createSpy(element.toString());
+        instance2.on(element, 'click', spy);
+        instance1.trigger('click');
+        instance2.off(element, 'click');
```
Comment (giuseppeg) [inline test/spec/events_spec.js]: probably overkill to also `off` here

**Comment by phuu** [comment]
Sorry @giuseppeg — looks like v2 isn't going to happen, will close this. Thanks for all your work.

**Comment by giuseppeg** [comment]
@phuu no problem, thank you guys for open sourcing the framework in the first place :)

**Pull request closed without merging**